app/src/main/java/at/tugraz/ist/swe/DrawArea.java

```diff
@@ -33,16 +34,18 @@ protected void onDraw(Canvas canvas) {
 
 
     }
-
     public boolean onTouchEvent(MotionEvent event) {
+
         paintingTool.handleEvent(event);
         invalidate();
-        oldBitmap = getBitmap();
-
-        return false;
+        if(event.getAction() != MotionEvent.ACTION_MOVE)
+        {
+            oldBitmap = getBitmap();
+            paintingTool.cleanUp();
+        }
+        return true;
     }
-
-    private Bitmap getBitmap()
+    public Bitmap getBitmap()
```

Comment:
Rename getBitmap into createBitmap [DO not forget to adjust the function calls inside onTouchEvent etc. which use getBitmap. Should be createBitmap then.](make it private again) and create a new function which is public with the name getBitmap which returns oldBitmap.